src/main/java/com/yahoo/bullet/storm/DSLSpout.java

```diff
@@ -0,0 +1,136 @@
+import com.yahoo.bullet.record.BulletRecord;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.storm.spout.SpoutOutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.IRichSpout;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Values;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public class DSLSpout extends ConfigComponent implements IRichSpout {
+    private static final long serialVersionUID = 9218045272408135524L;
+    private static final Long DUMMY_ID = 42L;
+
+    private SpoutOutputCollector collector;
+    private TopologyContext context;
+    private BulletDSLConfig bulletDSLConfig;
+
+    // Exposed for testing only
+    @Getter(AccessLevel.PACKAGE)
+    private BulletConnector connector;
+
+    @Getter(AccessLevel.PACKAGE)
+    private BulletRecordConverter converter;
+
+    /**
+     * Creates a DSLSpout with a {@link BulletStormConfig} constructed from a configuration file. This constructor is
+     * used in the {@link Topology} class.
+     *
+     * @param args A list of arguments where the first argument is expected to be the path to a configuration file.
+     */
+    public DSLSpout(List<String> args) {
+        this(new BulletStormConfig(args.get(0)));
+    }
+
+    /**
+     * Creates a DSLSpout with a given {@link BulletStormConfig}.
+     *
+     * @param config The non-null BulletStormConfig to use. It should contain the settings to initialize a BulletConnector and a BulletRecordConverter.
+     */
+    public DSLSpout(BulletStormConfig config) {
+        super(config);
+        bulletDSLConfig = new BulletDSLConfig(config);
+        connector = BulletConnector.from(bulletDSLConfig);
+        converter = BulletRecordConverter.from(bulletDSLConfig);
+    }
+
+    @Override
+    public void open(Map conf, TopologyContext context, SpoutOutputCollector collector) {
+        this.collector = collector;
+        this.context = context;
+    }
+
+    @Override
+    public void activate() {
+        try {
+            connector.initialize();
+        } catch (BulletDSLException e) {
+            throw new RuntimeException("Could not activate DSLSpout.", e);
+        }
+        log.info("DSLSpout activated");
+    }
+
+    @Override
+    public void deactivate() {
+        try {
+            connector.close();
+        } catch (Exception e) {
+            log.error("Could not close BulletConnector.", e);
+        }
+        log.info("DSLSpout deactivated");
+    }
+
+    @Override
+    public void nextTuple() {
+        List<Object> objects = Collections.emptyList();
+        try {
+            objects = connector.read();
+        } catch (BulletDSLException e) {
+            log.error("Could not read from BulletConnector.", e);
+        }
+        long time = System.currentTimeMillis();
```

Comment:
This time is a in-between time from reading to converting. Can we make it pre-emission after all conversion? Or pre-reading? Or get a new time per emission?